src/lib.rs

```diff
@@ -774,76 +712,106 @@ impl Docker {
     }
 
     /// Returns version information associated with the docker daemon
-    pub fn version(&self) -> Result<Version> {
-        let raw = self.get("/version")?;
-        Ok(serde_json::from_str::<Version>(&raw)?)
+    pub fn version(&self) -> impl Future<Item = Version, Error = Error> {
+        self.get_json("/version")
     }
 
     /// Returns information associated with the docker daemon
-    pub fn info(&self) -> Result<Info> {
-        let raw = self.get("/info")?;
-        Ok(serde_json::from_str::<Info>(&raw)?)
+    pub fn info(&self) -> impl Future<Item = Info, Error = Error> {
+        self.get_json("/info")
     }
 
     /// Returns a simple ping response indicating the docker daemon is accessible
-    pub fn ping(&self) -> Result<String> {
+    pub fn ping(&self) -> impl Future<Item = String, Error = Error> {
         self.get("/_ping")
     }
 
     /// Returns an interator over streamed docker events
-    pub fn events(
-        &self,
-        opts: &EventsOptions,
-    ) -> Result<Vec<Event>> {
+    pub fn events(&self, opts: &EventsOptions) -> impl Stream<Item = Event, Error = Error> {
```

Comment:
😍 good choice. streams are an awesome choice. Doesn't have to happen here but I just noticed the pre-existing typo in this rustdoc: `interator` -> `iterator`. it was both incorrect and a typo. we may just want to say something like `Returns a stream of docker events`. The docs for this crate could use some polish but that can definately be a followup